pkgs/development/libraries/openssl/default.nix

```diff
@@ -105,6 +105,10 @@ let
     '';
 
     postFixup = ''
+      # fixes cross-compilation to pick the right perl,
+      # since perl is also in nativeBuildInputs
+      (sed -i -e '1 s!^.*$!${perl}/bin/perl!' $bin/bin/c_rehash)
```

Comment:
I didn't think this would be a problem now that #49608 is merged. Do you know where this comes from? It also looks like openssl supports passing in HASHBANGPERL to do this explicity.

Reply:
`HASHBANGPERL` is only supported in openssl 1.1, but we still use openssl 1.0.2 in most places.

Reply:
Also #49608 was reverted again in staging by @LnL7

Reply:
Yeah, the stdenv was broken. Since this is the second time I think that change has a bit too much impact to merge into staging directly, we should run it through a separate jobset first to get an overview of the breakages.

Reply:
I probably don't need this change to make cross-compiling rust packages work. So I will split it of probably later.